src/odemis/gui/cont/views.py

```diff
@@ -325,31 +404,55 @@ def _on_chamber_state(self, state):
             # sample have probably nothing in common with this new sample
             self._data_model.stage_history.value = self._data_model.stage_history.value[-1:]
 
+    def _update_ovv(self):
+        """ Update the overview image with the currently active stream. """
+        if self._data_model.streams.value:
+            for s in self._data_model.streams.value:
+                if s.is_active.value and s.image.value is not None:
+                    img = s.image.value
+                    if isinstance(s, acqstream.OpticalStream):
+                        opt = img
+                        sem = None
+                    if isinstance(s, acqstream.EMStream):
+                        sem = img
+                        opt = None
+
+                    # Update optical or sem image, depending on which stream is active
+                    if opt is not None:
```

Comment:
That if block seems redundant with the previous one. Just merge them together, and drop the `opt` & `sem` variables.